railties/test/application/middleware_test.rb

```diff
@@ -161,6 +161,13 @@ def app
       assert_not_includes middleware, "ActionDispatch::HostAuthorization"
     end
 
+    test "ActionDispatch::HostAuthorization is included if hosts is set to IPAddr" do
+      add_to_config "config.hosts = IPAddr.new('0.0.0.0/0')"
```

Comment:
Good thinking setting this to an `IPAddr`, so it won't accidentally pass via `String#empty?`. :+1: